desk/src/composables/router.ts

```diff
@@ -0,0 +1,23 @@
+import { reactive, watch } from "vue"
+import { useRoute } from "vue-router"
+
+/**
+ * @returns reactive object with route params as strings
+ */
+export function useRouteParamsAsStrings() {
```

Comment:
VueUse has a [`useRouteParams` composable](https://vueuse.org/router/useRouteParams/#userouteparams) that does this per-key as well, and might be useful instead of re-defining it here

Reply:
That returns `string | string[]` too. Here I want the string params so did not use it. Also, want to keep dependencies minimum unless absolutely necessary 🙈

Reply:
> I don't think you'll need to explicitly-cast it to string like that, you should be able to do it programmatically:

Agreed but I will have to do this for all route params. I am accessing a bunch of them in code: module, doctype, id, workspace, etc across 2 files. Writing a composable just seemed cleaner here.

> That's fair, but VueUse has been developed by a bunch of the core contributors of Vue and it's both maintained and well-tested. If you look at the [test cases](https://github.com/vueuse/vueuse/blob/main/packages/router/useRouteParams/index.test.ts?rgh-link-date=2024-06-21T12%3A43%3A45Z) for that composable, it covers a lot more things than the composable defined here. Which means that you may need to revisit this composable on an edge case.

I agree with this too. I just don't need that level of complexity/cases right now. If I find the current composable getting more complex/ugly during the rewrite ahead, I will switch instead of re-engineering stuff.

Reply:
That's fair enough